spyderlib/plugins/configdialog.py

```diff
@@ -903,77 +916,337 @@ def _save_lang(self):
+
+        # Add the config settings, based on the current one.
+        custom_names.append(custom_name)
+        self.set_option('custom_names', custom_names)
+        for key in syntaxhighlighters.COLOR_SCHEME_KEYS:
+            name = "{0}/{1}".format(custom_name, key)
+            default_name = "{0}/{1}".format(self.current_scheme, key).lower()
+            option = self.get_option(default_name)
+            self.set_option(name, option)
+
+        # Now they need to be loaded! how to make a partial load_from_conf?
+        dlg = self.scheme_editor_dialog
+        dlg.add_color_scheme_stack(custom_name)
+        dlg.set_scheme(custom_name)
+        self.load_from_conf()
+
+        if dlg.exec_():
+            self.update_combobox()
+            index = len(names) + custom_names.index(custom_name) - 1 
+            self.schemes_combo.setCurrentIndex(index)
+        else:
+            # Delete the config ....
+            custom_names.remove(custom_name)
+            self.set_option('custom_names', custom_names)
+            dlg.delete_color_scheme_stack(custom_name)
+
+    def edit_scheme(self):
+        """Edit current scheme."""
+        dlg = self.scheme_editor_dialog
+        dlg.set_scheme(self.current_scheme)
+
+        if dlg.exec_():
+            pass
+
+    def delete_scheme(self):
+        """Deletes the currently selected custom color scheme."""
+        scheme_name = self.current_scheme
+
+        # Put the combobox in Spyder by default, when deleting a scheme
+        names = [n.lower() for n in self.get_option('names')]
+        self.set_scheme('Spyder')
+        self.schemes_combo.setCurrentIndex(names.index('spyder'))
+        self.set_option('selected', 'Spyder')
+
+        # Delete from custom_names
+        custom_names = self.get_option('custom_names')
+        custom_names.remove(scheme_name)
+        self.set_option('custom_names', custom_names)
+
+        self.update_combobox()
+        self.update_preview()
+
+    def set_scheme(self, scheme_name):
+        """
+        Helper method that set the current stack in the dialog to the
+        scheme with name 'schema_name'.
+        """
+        dlg = self.scheme_editor_dialog
+        dlg.set_scheme(scheme_name)
+
+    @Slot(str)
+    def reset_to_default(self):
+        """Restore initial values for default color schemes."""
+        # Checks that this is indeed a +default scheme
+        scheme = self.current_scheme.lower()
+        for key in syntaxhighlighters.COLOR_SCHEME_KEYS:
+            option = "{0}/{1}".format(scheme, key)
+            value = CONF.get_default(self.CONF_SECTION, option)
+            self.set_option(option, value)
+        self.load_from_conf()
+
+    # Import/Export
+    # -------------------------------------------------------------------------
+    def import_scheme(self):
+        """
+        TODO:
+        """
+        pass
+
+    def export_scheme(self):
+        """
+        TODO:
+        """
+        pass
+

```

Comment:
Yep, I agree with @Nodd. Please remove these useless methods. You know where they need to belong in the future, so there's no need to add them for now.